test/test_manifest.rb

```diff
@@ -507,8 +507,8 @@ def teardown
     manifest.find("coffee.js").each do |asset|
       paths << asset.logical_path
     end
-    assert paths.include?("coffee.js")
-    assert !paths.include?("coffee/index.js")
+    assert paths.include?("coffee/index.js")
+    assert !paths.include?("coffee.js")
```

Comment:
This broke Basecamp since they are using this patter:

```
 config.assets.precompile += %w(
    libraries.js
    all.css all.js
    desktop.css desktop.js
    tablet.css tablet.js
    phone.css phone.js
    native.css native.js
    email.css
  )
```

All these are /index* setups.

So I guess we should implement that `javascript_tag "foo.js" -> resolve("foo.js") -> manifest["foo/index.js"]` path.

cc @dhh